src/lib/mixer/mixer_multirotor.cpp

```diff
@@ -146,157 +188,202 @@ MultirotorMixer::from_text(Mixer::ControlCallback control_cb, uintptr_t cb_handl
-	float roll_pitch_scale = 1.0f;	// scale for demanded roll and pitch
-	float delta_out_max = max_out - min_out; // distance between the two extrema
-
-	// If the difference between the to extrema is smaller than 1.0, the boost can safely unsaturate a motor if needed
-	// without saturating another one.
-	// Otherwise, a scaler is computed to make the distance between the two extrema exacly 1.0 and the boost
-	// value is computed to maximize the roll-pitch control.
-	//
-	// Note: thrust boost is computed assuming thrust_scale==1 for all motors.
-	// On asymmetric platforms, some motors have thrust_scale<1,
-	// which may result in motor saturation after thrust boost is applied
-	// TODO: revise the saturation/boosting strategy
-	if (delta_out_max <= 1.0f) {
-		if (min_out < 0.0f) {
-			boost = -min_out;
-
-		} else if (max_out > 1.0f) {
-			boost = -(max_out - 1.0f);
-		}
+	for (unsigned i = 0; i < _rotor_count; i++) {
+		outputs[i] += k1 * delta_outputs[i];
+	}
+
+	// Compute the desaturation gain again based on the updated outputs.
+	// In most cases it will be zero. It won't be if max(outputs) - min(outputs) > max_output - min_output.
+	// In that case adding 0.5 of the gain will equilibrate saturations.
+	float k2 = 0.5f * compute_desaturation_gain(delta_outputs, outputs, sat_status, min_output, max_output);
 
-	} else {
-		roll_pitch_scale = 1.0f / (delta_out_max);
-		boost = 1.0f - ((max_out - thrust) * roll_pitch_scale + thrust);
+	for (unsigned i = 0; i < _rotor_count; i++) {
+		outputs[i] += k2 * delta_outputs[i];
 	}
+}
 
-	if (!_airmode) {
-		// disable positive boosting if not in air-mode
-		// boosting can only be positive when min_out < 0.0
-		// roll_pitch_scale is reduced accordingly
-		if (boost > 0.0f) {
-			roll_pitch_scale = thrust / (thrust - min_out);
-			boost = 0.0f;
-		}
+void MultirotorMixer::mix_airmode_rp(float roll, float pitch, float yaw, float thrust, float *outputs)
+{
+	// Airmode for roll and pitch, but not yaw
+
+	// Mix without yaw
+	for (unsigned i = 0; i < _rotor_count; i++) {
+		outputs[i] = roll * _rotors[i].roll_scale +
+			     pitch * _rotors[i].pitch_scale +
+			     thrust * _rotors[i].thrust_scale;
+
+		// Thrust will be used to unsaturate if needed
+		_tmp_array[i] = _rotors[i].thrust_scale;
 	}
 
-	// capture saturation
-	if (min_out < 0.0f) {
-		_saturation_status.flags.motor_neg = true;
+	minimize_saturation(_tmp_array, outputs, _saturation_status);
+
+	// Mix yaw independently
+	mix_yaw(yaw, outputs);
+}
+
+void MultirotorMixer::mix_airmode_rpy(float roll, float pitch, float yaw, float thrust, float *outputs)
+{
+	// Airmode for roll, pitch and yaw
+
+	// Do full mixing
+	for (unsigned i = 0; i < _rotor_count; i++) {
+		outputs[i] = roll * _rotors[i].roll_scale +
+			     pitch * _rotors[i].pitch_scale +
+			     yaw * _rotors[i].yaw_scale +
+			     thrust * _rotors[i].thrust_scale;
+
+		// Thrust will be used to unsaturate if needed
+		_tmp_array[i] = _rotors[i].thrust_scale;
```

Comment:
I am trying to understand the new mixing method. 
For `mix_airmode_rpy`, only thrust will be used to unsaturate. I don't really get why thrust_scale is delta_output?

Reply:
Or put it differently: what exactly should delta_output represent?

Reply:
never mind got it. But the name is somewhat confusing, given that it is not a delta

Reply:
What name would you suggest? Would `desaturation_vector` work?

Reply:
yep, thats sounds better. `delta` sounds like the difference between two values

Reply:
done